src/tsolvers/egraph/EnodeStore.cc

```diff
@@ -94,6 +94,8 @@ bool EnodeStore::needsEnode(PTRef tr) const {
         return true;
     } else if (logic.isVar(tr) and not logic.hasSortBool(tr)) {
         return true;
+    } else if (logic.yieldsSortUninterpreted(tr) or (logic.hasArrays() and logic.isArraySort(logic.getSortRef(tr)))) {
```

Comment:
`logic.yieldsSortUninterpreted(tr) ` is checked below, does not have to be added.